broker-util/oo-admin-ctl-app

```diff
@@ -121,13 +121,53 @@ reply = ResultIO.new
 begin
   case command
   when "start"
-    reply.append app.start
+    if cartridge
+      begin
+        ci = app.component_instances.find_by(cartridge_name: cartridge)
+      rescue Exception => e
+        puts "Cartridge #{cartridge} not found in the application #{app.name}"
+        exit 1
+      end
+      reply.append app.start(cartridge)
+    else
+      reply.append app.start
+    end
   when "stop"
-    reply.append app.stop  
+    if cartridge
+      begin
+        ci = app.component_instances.find_by(cartridge_name: cartridge)
+      rescue Exception => e
+        puts "Cartridge #{cartridge} not found in the application #{app.name}"
+        exit 1
+      end
+      reply.append app.stop(cartridge)
+    else
+      reply.append app.stop
+    end
   when "force-stop"
-    reply.append app.stop(nil, true)
+    if cartridge
+      begin
+        ci = app.component_instances.find_by(cartridge_name: cartridge)
+      rescue Exception => e
+        puts "Cartridge #{cartridge} not found in the application #{app.name}"
+        exit 1
+      end
+      reply.append app.stop(cartridge, true)
+    else
+      reply.append app.stop(nil, true)
+    end
   when "restart"
-    reply.append app.restart  
+    if cartridge
+      begin
+        ci = app.component_instances.find_by(cartridge_name: cartridge)
+      rescue Exception => e
+        puts "Cartridge #{cartridge} not found in the application #{app.name}"
+        exit 1
```

Comment:
See if you can find a reasonable way to avoid all the duplicate code.  You could simply move the begin/rescue bits to a function or you could get fancier with abstract all of it so that all you need to do is tell the function to start/stop/restart and it would do everything here.